weave

```diff
@@ -813,6 +813,16 @@ dns_args() {
     fi
 }
 
+rewrite_hosts() {
+    path=$(docker inspect -f '{{.HostsPath}}' $CONTAINER)
+    name=$(docker inspect -f '{{.Config.Hostname}}' $CONTAINER)
+    containers_path="$(dirname $(dirname "$path"))"
+    docker run --rm \
+        -v "$containers_path":"$containers_path" \
+        --entrypoint=/bin/sh \
+        $EXEC_IMAGE -c "/usr/bin/weavehosts $path $name $ALL_CIDRS"
```

Comment:
> I see. Do we still need the volume mount in the proxy then?

Not sure but probably not. I'll investigate if there are more usages.

> Also, couldn't we mount just the specific file instead of the dir? 

Not sure about how these Docker mounts work, but probably not, as we `delete("/etc/hosts")` and then `create("/etc/hosts")`, so we probably need the whole dir for creating the new hosts.

> And perhaps mount it in a fixed location - mounting it in some path-which-could-be-anywhere seems kinda dangerous.

If we mount `<some_path>/hosts` at `/tmp/<some_other_path>`, what should `<some_other_path>` be? Not sure that using a fixed path is safer than a _"random"_ path...

> Also, why is the entrypoint `/bin/sh` instead of `/usr/bin/weavehosts`? Seems like an unnecessary indirection.

Probably. I will change it.